Mathlib/Topology/Algebra/Module/StrongTopology.lean

```diff
@@ -208,6 +213,20 @@ theorem tendsto_iff_tendstoUniformlyOn {ι : Type*} {p : Filter ι} [UniformSpac
   rw [(embedding_coeFn σ F 𝔖).tendsto_nhds_iff, UniformOnFun.tendsto_iff_tendstoUniformlyOn]
   rfl
 
+variable {𝔖₁ 𝔖₂ : Set (Set E)}
+
+theorem uniformSpace_mono [UniformSpace F] [UniformAddGroup F] (h : 𝔖₂ ⊆ 𝔖₁) :
+    instUniformSpace σ F 𝔖₁ ≤ instUniformSpace σ F 𝔖₂ := by
+  simp_rw [uniformSpace_eq]
+  exact UniformSpace.comap_mono (UniformOnFun.mono (le_refl _) h)
+
+theorem topologicalSpace_mono [TopologicalSpace F] [TopologicalAddGroup F] (h : 𝔖₂ ⊆ 𝔖₁) :
+    instTopologicalSpace σ F 𝔖₁ ≤ instTopologicalSpace σ F 𝔖₂ := by
+  letI := TopologicalAddGroup.toUniformSpace F
+  haveI : UniformAddGroup F := comm_topologicalAddGroup_is_uniform
+  simp_rw [← uniformity_toTopologicalSpace_eq]
+  exact UniformSpace.toTopologicalSpace_mono (uniformSpace_mono σ F h)
```

Comment:
Is it significantly more annoying to allow the uniform and topological structures to vary, as in `UniformOnFun.mono`? I would suggest doing it while you're at it, but if there's anything annoying we can also leave this for later (I suspect we will need it at some point).

Reply:
This is kinda annoying since we don't have the `UniformSpace` arguments are not named, the statement becomes the old @-hell or we have to introduce more local notation. I think naming all `UniformSpace` arguments is cleaner, but there seems some funny problem with the order of arguments (Lean seems to not like named instances coming after explicit arguments).